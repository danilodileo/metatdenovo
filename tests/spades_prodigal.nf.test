nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    profile "test_spades_prodigal"

    test("-profile test_spades_prodigal") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_metatdenovo_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // Important files with unstable content
                    path("${params.outdir}/samtools/spades.SAMPLE1_PE.flagstat").readLines().size(),
                    path("${params.outdir}/summary_tables/spades.prodigal.counts.tsv.gz").linesGzip.findAll { l -> l.contains('orf') },
                    path("${params.outdir}/summary_tables/spades.prodigal.overall_stats.tsv.gz").linesGzip.size(),
                    path("${params.outdir}/summary_tables/spades.prodigal.overall_stats.tsv.gz").linesGzip.findAll { l -> l.contains('n_feature_count') },
                ).match() },
                { assert path("${params.outdir}/spades/spades.spades.log").readLines().any { l -> l.contains('======= SPAdes pipeline finished.') } },
                { assert path("${params.outdir}/spades/spades.assembly.gfa.gz").linesGzip.size() > 10000 },
                { assert path("${params.outdir}/spades/spades.transcripts.fa.gz").linesGzip.size() > 3500 },
                { assert path("${params.outdir}/samtools/spades.SAMPLE1_PE.idxstats").readLines().size() > 1700 },
                { assert path("${params.outdir}/prodigal/spades.prodigal.gff.gz").linesGzip.findAll { l -> l.contains('CDS') }.size() > 3000 },
                { assert path("${params.outdir}/summary_tables/spades.prodigal.counts.tsv.gz").linesGzip.size() > 3000 },
            )
        }
    }
}
