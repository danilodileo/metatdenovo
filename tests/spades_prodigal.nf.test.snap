{
    "-profile test_spades_prodigal": {
        "content": [
            46,
            {
                "BBMAP_ALIGN": {
                    "bbmap": 39.18,
                    "samtools": 1.21,
                    "pigz": 2.8
                },
                "BBMAP_INDEX": {
                    "bbmap": 39.18
                },
                "CAT_FASTQ": {
                    "cat": 9.5
                },
                "COLLECT_FEATURECOUNTS": {
                    "R": "4.2.3",
                    "dplyr": "1.1.2",
                    "dtplyr": "1.3.1",
                    "data.table": "1.14.8"
                },
                "COLLECT_STATS": {
                    "R": "4.2.3",
                    "dplyr": "1.1.2",
                    "readr": "2.1.4",
                    "purrr": "1.0.1",
                    "tidyr": "1.3.0",
                    "stringr": "1.5.0"
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "FEATURECOUNTS_CDS": {
                    "subread": "2.0.6"
                },
                "FORMATSPADES": {
                    "gzip": "1.11"
                },
                "FORMAT_PRODIGAL_GFF": {
                    "gzip": "1.11"
                },
                "PIGZ_PE_READS_FWD": {
                    "pigz": 2.8
                },
                "PIGZ_PE_READS_REV": {
                    "pigz": 2.8
                },
                "PRODIGAL_MODULE": {
                    "prodigal": "2.6.3",
                    "pigz": 2.6
                },
                "SAMTOOLS_FLAGSTAT": {
                    "samtools": 1.21
                },
                "SAMTOOLS_IDXSTATS": {
                    "samtools": 1.21
                },
                "SAMTOOLS_INDEX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_SORT": {
                    "samtools": 1.21
                },
                "SAMTOOLS_STATS": {
                    "samtools": 1.21
                },
                "SEQTK_MERGEPE": {
                    "seqtk": "1.4-r122"
                },
                "SPADES": {
                    "spades": "4.1.0"
                },
                "TRANSRATE": {
                    "transrate": "1.0.3"
                },
                "TRIMGALORE": {
                    "trimgalore": "0.6.10",
                    "cutadapt": 4.9,
                    "pigz": 2.8
                },
                "UNPIGZ_CONTIGS": {
                    "pigz": "2.3.4"
                },
                "UNPIGZ_GFF": {
                    "pigz": "2.3.4"
                },
                "WRITESPADESYAML": {
                    "writespadesyaml": "4.2.25(1)-release"
                },
                "Workflow": {
                    "nf-core/metatdenovo": "v1.2.1dev"
                }
            },
            [
                "bbmap",
                "bbmap/bbmap",
                "bbmap/bbmap/SAMPLE1_PE.bbmap.log",
                "bbmap/bbmap/SAMPLE2_PE.bbmap.log",
                "bbmap/bbmap/SAMPLE3_PE_1.bbmap.log",
                "featurecounts",
                "featurecounts/SAMPLE1_PE.featureCounts.tsv",
                "featurecounts/SAMPLE1_PE.featureCounts.tsv.summary",
                "featurecounts/SAMPLE2_PE.featureCounts.tsv",
                "featurecounts/SAMPLE2_PE.featureCounts.tsv.summary",
                "featurecounts/SAMPLE3_PE_1.featureCounts.tsv",
                "featurecounts/SAMPLE3_PE_1.featureCounts.tsv.summary",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/BETA-multiqc.parquet",
                "multiqc/multiqc_data/llms-full.txt",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_metatdenovo_software_mqc_versions.yml",
                "prodigal",
                "prodigal/spades.prodigal.faa.gz",
                "prodigal/spades.prodigal.fna.gz",
                "prodigal/spades.prodigal.gff.gz",
                "prodigal/spades.prodigal_all.txt.gz",
                "prodigal/spades.prodigal_format.gff.gz",
                "samtools",
                "samtools/SAMPLE1_PE.sorted.bam",
                "samtools/SAMPLE1_PE.sorted.bam.bai",
                "samtools/SAMPLE2_PE.sorted.bam",
                "samtools/SAMPLE2_PE.sorted.bam.bai",
                "samtools/SAMPLE3_PE_1.sorted.bam",
                "samtools/SAMPLE3_PE_1.sorted.bam.bai",
                "samtools/spades.SAMPLE1_PE.flagstat",
                "samtools/spades.SAMPLE1_PE.idxstats",
                "samtools/spades.SAMPLE2_PE.flagstat",
                "samtools/spades.SAMPLE2_PE.idxstats",
                "samtools/spades.SAMPLE3_PE_1.flagstat",
                "samtools/spades.SAMPLE3_PE_1.idxstats",
                "spades",
                "spades/spades.assembly.gfa.gz",
                "spades/spades.spades.log",
                "spades/spades.transcripts.fa.gz",
                "spades/spades.yaml",
                "summary_tables",
                "summary_tables/spades.prodigal.counts.tsv.gz",
                "summary_tables/spades.prodigal.overall_stats.tsv.gz",
                "transrate",
                "transrate/spades_assemblies_mqc.csv",
                "transrate/versions.yml",
                "trimgalore",
                "trimgalore/SAMPLE1_PE_1.fastq.gz_trimming_report.txt",
                "trimgalore/SAMPLE1_PE_2.fastq.gz_trimming_report.txt",
                "trimgalore/SAMPLE2_PE_1.fastq.gz_trimming_report.txt",
                "trimgalore/SAMPLE2_PE_2.fastq.gz_trimming_report.txt",
                "trimgalore/SAMPLE3_PE_1_1.fastq.gz_trimming_report.txt",
                "trimgalore/SAMPLE3_PE_1_2.fastq.gz_trimming_report.txt",
                "trimgalore/fastqc",
                "trimgalore/fastqc/SAMPLE1_PE_1_val_1_fastqc.html",
                "trimgalore/fastqc/SAMPLE1_PE_1_val_1_fastqc.zip",
                "trimgalore/fastqc/SAMPLE1_PE_2_val_2_fastqc.html",
                "trimgalore/fastqc/SAMPLE1_PE_2_val_2_fastqc.zip",
                "trimgalore/fastqc/SAMPLE2_PE_1_val_1_fastqc.html",
                "trimgalore/fastqc/SAMPLE2_PE_1_val_1_fastqc.zip",
                "trimgalore/fastqc/SAMPLE2_PE_2_val_2_fastqc.html",
                "trimgalore/fastqc/SAMPLE2_PE_2_val_2_fastqc.zip",
                "trimgalore/fastqc/SAMPLE3_PE_1_1_val_1_fastqc.html",
                "trimgalore/fastqc/SAMPLE3_PE_1_1_val_1_fastqc.zip",
                "trimgalore/fastqc/SAMPLE3_PE_1_2_val_2_fastqc.html",
                "trimgalore/fastqc/SAMPLE3_PE_1_2_val_2_fastqc.zip"
            ],
            [
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "versions.yml:md5,069fffe1c38bd4bbce26b40bcc449fb2"
            ],
            16,
            [
                "orf\tchr\tstart\tend\tstrand\tlength\tsample\tcount\ttpm"
            ],
            4,
            [
                "sample\tn_post_trimming\tidxs_n_mapped\tidxs_n_unmapped\tn_feature_count"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-11T20:38:22.526310018"
    }
}